typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -446,8 +435,48 @@ async function messageIsProcessed(
   destination: ChainName,
   message: DispatchedMessage,
 ): Promise<boolean> {
+  const ctx = { origin, destination, message };
+  // If we weren't able to get the receipt for message processing,
+  // try to read the state to ensure it wasn't a transient provider issue
+  log('Checking if message was received despite timeout', ctx);
+
   const destinationMailbox = core.getContracts(destination).mailbox.contract;
-  return destinationMailbox.delivered(message.id);
+
+  // Try a few times to see if the message has been processed --
+  // we've seen some intermittent issues when fetching state.
+  // This will throw if the message is found to have not been processed.
+  let isProcessed = false;
+  try {
```

Comment:
This code block could use cleaning up. I don't see why we can't just do:
```
try {
    isProcessed = await utils.retryAsync(async () => {
      if (!(await messageIsProcessed(app.core, origin, destination, message))) {
        throw error;
      }
    }, 3);
} catch (e) {};
```

Reply:
I don't understand this suggestion, this is non-equivalent.

Reply:
What's the difference?